pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -47,8 +57,32 @@
 @ThreadSafe
 public class BrokerReduceService {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(BrokerReduceService.class);
+
+  // brw -> Shorthand for broker reduce worker threads.
+  private static final String REDUCE_THREAD_NAME_FORMAT = "brw-%d";
+  protected static final int QUERY_RUNNER_THREAD_PRIORITY = 7;
+
+  private final ListeningExecutorService _reduceExecutorService;
+  private final int _maxReduceThreadsPerQuery;
+
+  public BrokerReduceService(PinotConfiguration config) {
+    _maxReduceThreadsPerQuery = config.getProperty(CommonConstants.Broker.CONFIG_OF_MAX_REDUCE_THREADS_PER_QUERY,
+        CommonConstants.Broker.MAX_REDUCE_THREADS_PER_QUERY);
+    LOGGER.info("Initializing BrokerReduceService with {} reduce threads.", _maxReduceThreadsPerQuery);
+
+    ThreadFactory reduceThreadFactory =
+        new ThreadFactoryBuilder().setDaemon(false).setPriority(QUERY_RUNNER_THREAD_PRIORITY)
+            .setNameFormat(REDUCE_THREAD_NAME_FORMAT).build();
+
+    // ExecutorService is initialized with numThreads sames availableProcessors.
+    ExecutorService delegate =
+        Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors(), reduceThreadFactory);
+    _reduceExecutorService = MoreExecutors.listeningDecorator(delegate);
```

Comment:
I don't think we need to use the `ListeningExecutorService` here, `ExecutorService` should be enough with lower overhead